### ISSUE TITLE
static mut bindings don't raise unused_mut diagnostic

### ISSUE THREAD

**Issue by Arnavion** [rust-lang/rust#68721]
```rust
#[deny(unused_mut)]
pub unsafe fn foo() {
    static mut X: std::sync::Once = std::sync::Once::new();
    X.call_once(|| { /**/ });
}
```

... [compiles fine.](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=eb6b3a8bd59dee6e53161934988de3e7) Using `let mut X` version fails as expected.

I can imagine it would be hard to do this for module-scoped `static`s since the mutable access could occur in any scope the value is visible to, but could it be done for function-scoped ones at least?

---

I discovered this in a crate with a bunch of function-scoped `Once`s, and just happened to notice some of them had been created with `static mut` and others with just `static`, indicating the `mut` was unnecessary even though the compiler didn't complain.

**Comment by jonas-schievink** [comment]
This probably happens because a `static mut` is fundamentally a different language construct than an immutable `static` or any `let`-binding (mutable or not), and the lint only looks at variable bindings.
